src/api/c/quiz3/main.c

```diff
@@ -15,6 +16,53 @@
 #define FILE_NAME_1 "/home/rosser/atsm/examples/c/syscalls/file/ifv/Makefile"
 #define FILE_NAME_2 "/home/rosser/atsm/examples/c/syscalls/file/ifv/Makefile.orig"
 
+void hex_to_image(char outputf[], int hexSize, char hexdump[])
+{
+	int fd;
+	char num [13];
+	int iter = 0;
+
+	/*Create the file with the given ouput filename*/
+	fd = open(outputf, O_RDWR | O_CREAT, S_IRUSR | S_IRGRP | S_IROTH);
```

Comment:
`fd` is never checked for the return value from the system call. Particularly bad idea inside a driver if it fails...